### PR TITLE
Feature/pelagos 4495 make output of submissions fields

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+use App\Form\PersonDatasetSubmissionType;
+use App\Repository\DatasetRepository;
+use Doctrine\Common\Collections\Collection;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * A controller for a Report of Research Groups and related Datasets.
+ *
+ * @return Response A Symfony Response instance.
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:18]: Shouldn't have a return statement here

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+class ReportDatasetSubmission extends ReportController
+{
+    /**
+     * The default action for Dataset Review.
+     *
+     * @param string            $udi               A UDI.
+     * @param DatasetRepository $datasetRepository The dataset repository.
+     *
+     * @throws NotFoundHttpException When the dataset does not exist.
+     *
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:29]: Extra line

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
+<?php
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:1]: Should this also have a link on the admin page?

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    /**
+     * The default action for Dataset Review.
+     *
+     * @param string            $udi               A UDI.
+     * @param DatasetRepository $datasetRepository The dataset repository.
+     *
+     * @throws NotFoundHttpException When the dataset does not exist.
+     *
+     *
+     * @Route("/overview/{udi}")
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:31]: No route name

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+use App\Entity\PersonDatasetSubmission;
+use App\Form\DatasetSubmissionType;
+use App\Form\PersonDatasetSubmissionType;
+use App\Repository\DatasetRepository;
+use Doctrine\Common\Collections\Collection;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * A controller for a Report of Research Groups and related Datasets.
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:16]: Can change class description

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 6]
```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace App\Controller\UI;
+
+use App\Entity\Dataset;
+use App\Entity\DatasetSubmission;
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:6]: Not used

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+use App\Entity\PersonDatasetSubmission;
+use App\Form\DatasetSubmissionType;
+use App\Form\PersonDatasetSubmissionType;
+use App\Repository\DatasetRepository;
+use Doctrine\Common\Collections\Collection;
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:11]: Not used

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+            'themeKeywords',
+            'placeKeywords',
+            'topicKeywords',
+            'spatialExtent',
+            'spatialExtentDescription',
+            'temporalExtentNilReasonType',
+            'temporalExtentDesc',
+            'temporalExtentBeginPosition',
+            'temporalExtentEndPosition',
+            'restrictions',
```
Comment (praneethpr) [inline src/Controller/UI/ReportDatasetSubmission.php:72]: The ticket's example csv shows that have Remotely hosted and cold storage attributes

File: src/Controller/UI/ReportDatasetSubmission.php (hunk 4 above)
Comment (mickel1138) [reply src/Controller/UI/ReportDatasetSubmission.php:31]: Route name is optional, it makes a "magic" route from the class, controller, action.

File: src/Controller/UI/ReportDatasetSubmission.php (hunk 8 above)
Comment (mickel1138) [reply src/Controller/UI/ReportDatasetSubmission.php:72]: That's old.

**Review by fightingtexasaggie — COMMENTED** [review]
This is minor, but I did notice spatial extent has no heading

File: src/Controller/UI/ReportDatasetSubmission.php (hunk 3 above)
Comment (mickel1138) [reply src/Controller/UI/ReportDatasetSubmission.php:1]: We'll let the PO decide, was not in the ticket.

**Review by praneethpr — DISMISSED** [review]
Tested and works fine

File: src/Controller/UI/ReportDatasetSubmission.php [hunk 9]
```diff
@@ -89,7 +89,7 @@ public function makeCSVAction(string $udi, DatasetRepository $datasetRepository)
             $label = $childView->vars['label'];
 
             if (empty($label)) {
-                continue;
+                $label = $childView->vars['name'] . ':';
```
Comment (fightingtexasaggie) [inline src/Controller/UI/ReportDatasetSubmission.php:92]: This is the only field that has a colon after it. Also, we should probably quote the label in the generated CSV.

Comment (fightingtexasaggie) [inline src/Controller/UI/ReportDatasetSubmission.php:92]:
```suggestion
                $label = '"' . $childView->vars['name'] . '"';
```

Comment (mickel1138) [reply src/Controller/UI/ReportDatasetSubmission.php]: Removed, also removed from others, not needed because it's not consistent.

**Review by praneethpr — APPROVED** [review]

**Review by fightingtexasaggie — APPROVED** [review]
looks good

**Pull request merged**